src/i18n.ts

```diff
@@ -0,0 +1,41 @@
+import i18n from "i18next";
+import { initReactI18next } from "react-i18next";
+
+import Backend from "i18next-http-backend";
+import LanguageDetector from "i18next-browser-languagedetector";
+import { format as date_fnz_tz_format } from "date-fns-tz";
+import enCA from "date-fns/locale/en-CA";
+import frCA from "date-fns/locale/fr-CA";
+// don't want to use this?
+// have a look at the Quick start guide
+// for passing in lng and translations on init
+
+i18n
+  // load translation using http -> see /public/locales (i.e. https://github.com/i18next/react-i18next/tree/master/example/react/public/locales)
+  // learn more: https://github.com/i18next/i18next-http-backend
+  .use(Backend)
+  // detect user language
+  // learn more: https://github.com/i18next/i18next-browser-languageDetector
+  .use(LanguageDetector)
+  // pass the i18n instance to react-i18next.
+  .use(initReactI18next)
+  // init i18next
+  // for all options read: https://www.i18next.com/overview/configuration-options
+  .init({
```

Comment:
Here is how I have set things up in the past to eager load the localizations:

```
import englishTranslations from "src/locales/en-CA.json";
import frenchTranslations from "src/locales/fr-CA.json";
...
const CONSTANTS = {
  ENGLISH_CODE: "en-CA",
  FRENCH_CODE: "en-FR",
}

const resources = {
  [CONSTANTS.ENGLISH_CODE]: {
    translation: englishTranslations,
  },
  [CONSTANTS.FRENCH_CODE]: {
    translation: frenchTranslations,
  },
};

i18next
  .use(detector)
  .use(initReactI18next)
  .init({
    resources,
    fallbackLng: CONSTANTS.ENGLISH_CODE,
  });
```

Reply:
The configuration on i18next help uses lazy loading and I copied from that. Is there a significant performance improvement if I change this?

Reply:
Lazy loading means:

* more dependencies
* a more complicated configuration
* slightly longer TTI as we can't do anything until the web app spins up, initiates the request, receives the translation and applies them.

Lazy loading reaps benefits when the resources being loaded are large and varied. If our localization files were thousands of lines and spanned 100 languages, then lazy loading would absolutely be appropriate. But when we're talking about a couple tiny files, the network overhead of the AJAX requests to fetch them means there is no value in going this way.